bug: 设置flaresolverr重建docker后报错

### 问题描述

如日志所述

### 复现步骤

webui设置选项为flaresolverr保存后重建docker compose

### 版本

v2.0.0

### 相关日志

```shell
[INFO] Starting granian (main PID: 1)
[INFO] Listening at: http://0.0.0.0:8000
[INFO] Spawning worker-1 with PID: 12
2026-04-10 11:58:02.654 | INFO     | app.main:lifespan:90 - application startup: service=grok2api python=3.13.13 platform=Linux
2026-04-10 11:58:02.664 | INFO     | app.main:lifespan:103 - account storage configured: backend=local target=/app/data/accounts.db
2026-04-10 11:58:02.686 | INFO     | app.dataplane.account.sync:bootstrap:67 - account runtime table bootstrapped: revision=10 account_count=276 pool_count=1
2026-04-10 11:58:02.687 | INFO     | app.dataplane.account:bootstrap:52 - account directory ready: size=276
2026-04-10 11:58:02.687 | INFO     | app.control.account.scheduler:start:51 - account refresh scheduler started: basic_interval_s=36000 super_interval_s=7200 heavy_interval_s=7200
2026-04-10 11:58:02.687 | INFO     | app.main:lifespan:154 - scheduler leader: pid=12 active_sync_s=3 idle_sync_s=30
[ERROR] Traceback (most recent call last):
  File "/opt/venv/lib/python3.13/site-packages/starlette/routing.py", line 694, in lifespan
    async with self.lifespan_context(app) as maybe_state:
               ~~~~~~~~~~~~~~~~~~~~~^^^^^
  File "/usr/local/lib/python3.13/contextlib.py", line 214, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.13/site-packages/fastapi/routing.py", line 153, in merged_lifespan
    async with original_context(app) as maybe_original_state:
               ~~~~~~~~~~~~~~~~^^^^^
  File "/usr/local/lib/python3.13/contextlib.py", line 214, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.13/site-packages/fastapi/routing.py", line 153, in merged_lifespan
    async with original_context(app) as maybe_original_state:
               ~~~~~~~~~~~~~~~~^^^^^
  File "/usr/local/lib/python3.13/contextlib.py", line 214, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.13/site-packages/fastapi/routing.py", line 153, in merged_lifespan
    async with original_context(app) as maybe_original_state:
               ~~~~~~~~~~~~~~~~^^^^^
  File "/usr/local/lib/python3.13/contextlib.py", line 214, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/app/app/main.py", line 160, in lifespan
    await get_proxy_directory()
  File "/app/app/control/proxy/__init__.py", line 212, in get_proxy_directory
    await _directory.load()
  File "/app/app/control/proxy/__init__.py", line 48, in load
    self._clearance_mode = ClearanceMode(cfg.get_str("proxy.clearance.mode", "none"))
                           ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/enum.py", line 727, in __call__
    return cls.__new__(cls, value)
           ~~~~~~~~~~~^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/enum.py", line 1204, in __new__
    raise ve_exc
ValueError: 'flaresolverr' is not a valid ClearanceMode
[ERROR] ASGI lifespan startup failed
Traceback (most recent call last):
  File "/opt/venv/lib/python3.13/site-packages/granian/asgi.py", line 26, in handle
    await self.callable(
    ...<3 lines>...
    )
  File "/opt/venv/lib/python3.13/site-packages/fastapi/applications.py", line 1135, in __call__
    await super().__call__(scope, receive, send)
  File "/opt/venv/lib/python3.13/site-packages/starlette/applications.py", line 107, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/opt/venv/lib/python3.13/site-packages/starlette/middleware/errors.py", line 151, in __call__
    await self.app(scope, receive, send)
  File "/opt/venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 103, in __call__
    await self.app(scope, receive, send)
  File "/opt/venv/lib/python3.13/site-packages/starlette/middleware/cors.py", line 77, in __call__
    await self.app(scope, receive, send)
  File "/opt/venv/lib/python3.13/site-packages/starlette/middleware/exceptions.py", line 49, in __call__
    await self.app(scope, receive, send)
  File "/opt/venv/lib/python3.13/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/opt/venv/lib/python3.13/site-packages/starlette/routing.py", line 716, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/opt/venv/lib/python3.13/site-packages/starlette/routing.py", line 725, in app
    await self.lifespan(scope, receive, send)
  File "/opt/venv/lib/python3.13/site-packages/starlette/routing.py", line 694, in lifespan
    async with self.lifespan_context(app) as maybe_state:
               ~~~~~~~~~~~~~~~~~~~~~^^^^^
  File "/usr/local/lib/python3.13/contextlib.py", line 214, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.13/site-packages/fastapi/routing.py", line 153, in merged_lifespan
    async with original_context(app) as maybe_original_state:
               ~~~~~~~~~~~~~~~~^^^^^
  File "/usr/local/lib/python3.13/contextlib.py", line 214, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.13/site-packages/fastapi/routing.py", line 153, in merged_lifespan
    async with original_context(app) as maybe_original_state:
               ~~~~~~~~~~~~~~~~^^^^^
  File "/usr/local/lib/python3.13/contextlib.py", line 214, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/opt/venv/lib/python3.13/site-packages/fastapi/routing.py", line 153, in merged_lifespan
    async with original_context(app) as maybe_original_state:
               ~~~~~~~~~~~~~~~~^^^^^
  File "/usr/local/lib/python3.13/contextlib.py", line 214, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/app/app/main.py", line 160, in lifespan
    await get_proxy_directory()
  File "/app/app/control/proxy/__init__.py", line 212, in get_proxy_directory
    await _directory.load()
  File "/app/app/control/proxy/__init__.py", line 48, in load
    self._clearance_mode = ClearanceMode(cfg.get_str("proxy.clearance.mode", "none"))
                           ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/enum.py", line 727, in __call__
    return cls.__new__(cls, value)
           ~~~~~~~~~~~^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/enum.py", line 1204, in __new__
    raise ve_exc
ValueError: 'flaresolverr' is not a valid ClearanceMode
[ERROR] Unexpected exit from worker-1
[INFO] Shutting down granian
```

--- a/README.md
+++ b/README.md
@@ -14,8 +14,9 @@
 <br>
 
 Grok2API 是一个基于 **FastAPI** 构建的 Grok 网关，支持将 Grok Web 能力以 OpenAI 兼容 API 的方式转换。核心特性：
-- OpenAI 兼容接口：`/v1/chat/completions`、`/v1/responses`、`/v1/images/generations`、`/v1/images/edits`、`/v1/videos`
-- 支持流式与非流式对话、显式思考输出、函数工具结构透传
+- OpenAI 兼容接口：`/v1/models`、`/v1/chat/completions`、`/v1/responses`、`/v1/images/generations`、`/v1/images/edits`、`/v1/videos`、`/v1/videos/{video_id}`、`/v1/videos/{video_id}/content`
+- Anthropic 兼容接口：`/v1/messages`
+- 支持流式与非流式对话、显式思考输出、函数工具结构透传，以及统一的 token / usage 统计
 - 支持多账号池、层级选号、失败反馈、额度同步与自动维护
 - 支持本地缓存图片、视频与本地代理链接返回
 - 支持文生图、图像编辑、文生视频、图生视频
@@ -30,28 +31,32 @@ flowchart LR
     Client["Clients\nOpenAI SDK / curl / Browser"] --> API["FastAPI App"]
 
     subgraph Products["Products"]
+        direction TB
         OpenAI["OpenAI APIs\n/v1/*"]
         Anthropic["Anthropic APIs\n/v1/messages"]
         Web["Web Products\n/admin /webui/*"]
     end
 
     subgraph Control["Control"]
+        direction TB
         Models["Model Registry"]
         Accounts["Account Services"]
         Proxies["Proxy Services"]
     end
 
     subgraph Dataplane["Dataplane"]
+        direction TB
+        Reverse["Reverse Protocol + Transport"]
         AccountDP["AccountDirectory"]
         ProxyDP["Proxy Runtime"]
-        Reverse["Reverse Protocol + Transport"]
     end
 
     subgraph Platform["Platform"]
-        Config["Config Snapshot"]
-        Auth["Auth"]
+        direction TB
         Tokens["Token Estimation"]
         Storage["Storage"]
+        Config["Config Snapshot"]
+        Auth["Auth"]
         Log["Logging"]
     end
 

--- a/app/control/proxy/__init__.py
+++ b/app/control/proxy/__init__.py
@@ -45,7 +45,7 @@ class ProxyDirectory:
         """Load proxy configuration from the current config snapshot."""
         cfg = get_config()
         self._egress_mode    = EgressMode(cfg.get_str("proxy.egress.mode", "direct"))
-        self._clearance_mode = ClearanceMode(cfg.get_str("proxy.clearance.mode", "none"))
+        self._clearance_mode = ClearanceMode.parse(cfg.get_str("proxy.clearance.mode", "none"))
 
         nodes: list[EgressNode]          = []
         resource_nodes: list[EgressNode] = []

--- a/app/control/proxy/models.py
+++ b/app/control/proxy/models.py
@@ -24,9 +24,18 @@ class EgressMode(StrEnum):
 
 
 class ClearanceMode(StrEnum):
-    NONE    = "none"     # no CF clearance required
-    MANUAL  = "manual"   # operator-supplied cf_cookies
-    MANAGED = "managed"  # maintained by FlareSolverr
+    NONE         = "none"         # no CF clearance required
+    MANUAL       = "manual"       # operator-supplied cf_cookies
+    FLARESOLVERR = "flaresolverr" # maintained by FlareSolverr
+
+    @classmethod
+    def parse(cls, value: str | "ClearanceMode") -> "ClearanceMode":
+        if isinstance(value, cls):
+            return value
+        normalized = str(value or "").strip().lower()
+        if not normalized:
+            return cls.NONE
+        return cls(normalized)
 
 
 class EgressNodeState(IntEnum):

--- a/app/control/proxy/providers/flaresolverr.py
+++ b/app/control/proxy/providers/flaresolverr.py
@@ -37,8 +37,8 @@ class FlareSolverrClearanceProvider:
         proxy_url:    str,
     ) -> ClearanceBundle | None:
         cfg = get_config()
-        mode = cfg.get_str("proxy.clearance.mode", "none")
-        if mode != ClearanceMode.MANAGED:
+        mode = ClearanceMode.parse(cfg.get_str("proxy.clearance.mode", "none"))
+        if mode != ClearanceMode.FLARESOLVERR:
             return None
         fs_url      = cfg.get_str("proxy.clearance.flaresolverr_url", "")
         timeout_sec = cfg.get_int("proxy.clearance.timeout_sec", 60)
@@ -58,7 +58,7 @@ class FlareSolverrClearanceProvider:
             return None
 
         return ClearanceBundle(
-            bundle_id    = f"managed:{affinity_key}",
+            bundle_id    = f"flaresolverr:{affinity_key}",
             cf_cookies   = result.get("cookies", ""),
             user_agent   = result.get("user_agent", ""),
             affinity_key = affinity_key,

--- a/app/control/proxy/providers/manual.py
+++ b/app/control/proxy/providers/manual.py
@@ -9,7 +9,7 @@ class ManualClearanceProvider:
 
     def build_bundle(self, *, affinity_key: str) -> ClearanceBundle | None:
         cfg = get_config()
-        mode = cfg.get_str("proxy.clearance.mode", "none")
+        mode = ClearanceMode.parse(cfg.get_str("proxy.clearance.mode", "none"))
         if mode != ClearanceMode.MANUAL:
             return None
         return ClearanceBundle(

--- a/docs/README.en.md
+++ b/docs/README.en.md
@@ -4,7 +4,7 @@
 [![FastAPI](https://img.shields.io/badge/FastAPI-0.119%2B-009688?logo=fastapi&logoColor=white)](https://fastapi.tiangolo.com/)
 [![Version](https://img.shields.io/badge/version-2.0.0.rc0-111827)](../pyproject.toml)
 [![License](https://img.shields.io/badge/license-MIT-16a34a)](../LICENSE)
-[![中文](https://img.shields.io/badge/中文-2563EB?logo=bookstack&logoColor=white)](../REAMDE.md)
+[![中文](https://img.shields.io/badge/中文-2563EB?logo=bookstack&logoColor=white)](../README.md)
 [![Project%20Docs](https://img.shields.io/badge/Project%20Docs-0F766E?logo=readthedocs&logoColor=white)](https://blog.cheny.me/blog/posts/grok2api)
 
 > [!NOTE]
@@ -13,8 +13,9 @@
 <br>
 
 Grok2API is a **FastAPI**-based Grok gateway that exposes Grok Web capabilities through OpenAI-compatible APIs. Core features:
-- OpenAI-compatible endpoints: `/v1/chat/completions`, `/v1/responses`, `/v1/images/generations`, `/v1/images/edits`, `/v1/videos`
-- Streaming and non-streaming chat, explicit reasoning output, and function-tool structure passthrough
+- OpenAI-compatible endpoints: `/v1/models`, `/v1/chat/completions`, `/v1/responses`, `/v1/images/generations`, `/v1/images/edits`, `/v1/videos`, `/v1/videos/{video_id}`, `/v1/videos/{video_id}/content`
+- Anthropic-compatible endpoint: `/v1/messages`
+- Streaming and non-streaming chat, explicit reasoning output, function-tool structure passthrough, and unified token / usage accounting
 - Multi-account pools, tier-aware selection, failure feedback, quota synchronization, and automatic maintenance
 - Local image/video caching and locally proxied media URLs
 - Text-to-image, image editing, text-to-video, and image-to-video support
@@ -29,28 +30,32 @@ flowchart LR
     Client["Clients\nOpenAI SDK / curl / Browser"] --> API["FastAPI App"]
 
     subgraph Products["Products"]
+        direction TB
         OpenAI["OpenAI APIs\n/v1/*"]
         Anthropic["Anthropic APIs\n/v1/messages"]
         Web["Web Products\n/admin /webui/*"]
     end
 
     subgraph Control["Control"]
+        direction TB
         Models["Model Registry"]
         Accounts["Account Services"]
         Proxies["Proxy Services"]
     end
 
     subgraph Dataplane["Dataplane"]
+        direction TB
+        Reverse["Reverse Protocol + Transport"]
         AccountDP["AccountDirectory"]
         ProxyDP["Proxy Runtime"]
-        Reverse["Reverse Protocol + Transport"]
     end
 
     subgraph Platform["Platform"]
-        Config["Config Snapshot"]
-        Auth["Auth"]
+        direction TB
         Tokens["Token Estimation"]
         Storage["Storage"]
+        Config["Config Snapshot"]
+        Auth["Auth"]
         Log["Logging"]
     end
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [project]
 name = "grok2api"
-version = "2.0.0.rc0"
+version = "2.0.1"
 description = "Grok2API rebuilt with FastAPI, fully aligned with the latest web call format. Supports streaming and non-streaming chat, image generation/editing, deep thinking, token pool concurrency, and automatic load balancing."
 readme = "README.md"
 requires-python = ">=3.13"

